### ISSUE TITLE
Setting Highest Quality or 1080p on downloads not working as it should

### ISSUE THREAD

**Issue by Tunant3** [FaisalUmair/udemy-downloader-gui#134]
Hi, using lates 1.6.0 windows x64. Setting downloads to highest quality or 1080p results in all videos downloaded at 720p even though 1080p is available for that particular course (Not available for all courses)

**Comment by FaisalUmair** [comment]
Can you link me to any such free course so that I can reproduce this issue.

**Comment by Tunant3** [comment]
It's not easy to find a free course with 1080p videos...will try

**Comment by Tunant3** [comment]
Hi, found a course with 1080p videos, "The Essentials of Acoustic Guitar Strumming Patterns", [https://www.udemy.com/strumming-mastery/learn/v4/content](url). Setting to highest or 1080p Will result in the correct download of 1080p videos. I can only reproduce the issue on paid courses. On those courses, I could use IDM to download 1080p version but now i get an error, "the server replies you don't have permission to download file", I guess they have change permissions on files.

**Comment by FaisalUmair** [comment]
Alright. Thanks for reporting.

I will see if I have a paid course which supports 1080p and try to reproduce the issue on those course.

**Comment by Saabbir** [comment]
Hello @FaisalUmair  I'm facing the same issue.

I have bought this -> https://www.udemy.com/course/modern-javascript/ course and I can watch it in 1080p but I can't download the course in 1080p using Udeler. It gets downloaded in 720p.

Here is a screenshot - http://prntscr.com/pfb1jz

If you can solve this issue that will be huge help mate!

Anyway, you've done great work on this project already. Thank you so much.

**Comment by rockus123** [comment]
Yes.  I have the same issue.  It only download 720p, regardless of settings.  Video is available in 1080p.

**Comment by pankpan** [comment]
I have the same issue too.
tried Auto, 1080p, Highest always get 720p.

**Comment by Paw2Pom** [comment]
Yes I have the similar issue.

I am not sure whether this post will help you find the solution?
https://medium.com/@s_54260/how-to-udemy-always-watch-videos-at-1080p-resolution-20d8f73ffaf1

Please help us on this issue, as downloading at 1080p will def make life easier as the video will become way more clearer to see.

Thank you!

**Comment by aradalvand** [comment]
Same issue here, any updates?

**Comment by dariusz22p** [comment]
same here (Mac) - no option to get 1080p

**Comment by anhtuank7c** [comment]
Same here (Mac) - I had updated the setting to download 1080p but no luck. Video course available to watch with 1080p resolution

**Comment by zyu0** [comment]
me too……how to download 1080p……
